Gems/Atom/Tools/AtomToolsFramework/Code/Include/AtomToolsFramework/Application/AtomToolsApplication.h

```diff
@@ -0,0 +1,111 @@
+
+        using Base = AzFramework::Application;
+
+        AtomToolsApplication(int* argc, char*** argv);
+
+        //////////////////////////////////////////////////////////////////////////
+        // AzFramework::Application
+        void CreateReflectionManager() override;
+        void Reflect(AZ::ReflectContext* context) override;
+        void RegisterCoreComponents() override;
+        AZ::ComponentTypeList GetRequiredSystemComponents() const override;
+        void CreateStaticModules(AZStd::vector<AZ::Module*>& outModules) override;
+        const char* GetCurrentConfigurationName() const override;
+        void StartCommon(AZ::Entity* systemEntity) override;
+        void Tick(float deltaOverride = -1.f) override;
+        void Stop() override;
+
+    protected:
+        //////////////////////////////////////////////////////////////////////////
+        // AssetDatabaseRequestsBus::Handler overrides...
+        bool GetAssetDatabaseLocation(AZStd::string& result) override;
+        //////////////////////////////////////////////////////////////////////////
+
+        //////////////////////////////////////////////////////////////////////////
+        // AzFramework::Application overrides...
+        void Destroy() override;
+        //////////////////////////////////////////////////////////////////////////
+
+        //////////////////////////////////////////////////////////////////////////
+        // AZ::ComponentApplication overrides...
+        void QueryApplicationType(AZ::ApplicationTypeQuery& appType) const override;
+        //////////////////////////////////////////////////////////////////////////
+
+        //////////////////////////////////////////////////////////////////////////
+        // AZ::UserSettingsOwnerRequestBus::Handler overrides...
+        void SaveSettings() override;
+        //////////////////////////////////////////////////////////////////////////
+
+        ////////////////////////////////////////////////////////////////////////
+        // EditorPythonConsoleNotificationBus::Handler overrides...
+        void OnTraceMessage(AZStd::string_view message) override;
+        void OnErrorMessage(AZStd::string_view message) override;
+        void OnExceptionMessage(AZStd::string_view message) override;
+        ////////////////////////////////////////////////////////////////////////
+
+        virtual AZStd::string_view GetBuildTargetName() {return targetName;};
+
+        virtual void LoadSettings();
+        virtual void UnloadSettings();
+        virtual void CompileCriticalAssets(const AZStd::vector<AZStd::string> &assetFiltersArray);
+        virtual void ProcessCommandLine(const AZ::CommandLine& commandLine);
+        virtual bool LaunchDiscoveryService();
+        virtual void StartInternal();
+
+        static void PyIdleWaitFrames(uint32_t frames);
+        void setTargetName(AZStd::string newTargetName);
+
+        AZStd::string targetName = "AtomTools";
```

Comment:
This variable name does not follow standard naming convention.  It should be m_targetName.  But you should not need the variable since the function can just return the string and be overridden by the derived classes.